tests/ship_client.cpp

```diff
@@ -97,37 +93,29 @@ int main(int argc, char* argv[]) {
          };
 
          while(num_requests--) {
-            rapidjson::StringBuffer request_sb;
-            rapidjson::PrettyWriter<rapidjson::StringBuffer> request_writer(request_sb);
-
-            request_writer.StartArray();
-               request_writer.String(request_result_types[num_requests%2].get_status_request.c_str());
-               request_writer.StartObject();
-               request_writer.EndObject();
-            request_writer.EndArray();
-
-            stream.write(boost::asio::buffer(request_type.json_to_bin(request_sb.GetString(), [](){})));
+            const eosio::chain::bytes get_status_bytes = abi.variant_to_binary("request",
+               fc::variants{request_result_types[num_requests%2].get_status_request, mvo()},
```

Comment:
I'm not quite understanding why we use a vector here:
`fc::variants{request_result_types[num_requests%2].get_status_request, mvo()}`

instead of:
`fc::variant{request_result_types[num_requests%2].get_status_request}`?

Reply:
It's a variant: type is either `get_status_request_v0` or `get_status_request_v1` (the test just alternates between the two), but both types are just empty structs so just empty `mvo()`

Reply:
Thanks Matt!